src/solver/variable/info.h

```diff
@@ -575,6 +575,19 @@ struct VariableAccessor<ResultsT, Category::singleColumn /* The default */>
         }
     }
 
+    template<class VCardType>
+    static SurveyResults::CaptionType getCaption(SurveyResults& results)
+    {
+        if (VCardType::categoryDataLevel & Category::bindingConstraint)
+        {
+            // In case the current output variable is associated to a BC, its caption is
+            // already set in the output variable class.
+            return results.variableCaption;
+        }
```

Comment:
I don't have access to the code, however for thermal cluster details, the situation is identical : the column name depends on the object. We should check how it's done, maybe the implementation is more simple.

Reply:
Ok, so there is a template
```cpp
template<class ResultsT, int ColumnCountT>
struct VariableAccessor
```
and 3 (partial) specializations
```cpp
template<class ResultsT>
struct VariableAccessor<ResultsT, Category::dynamicColumns>
// ...
template<class ResultsT>
struct VariableAccessor<ResultsT, Category::singleColumn /* The default */>
// ...
template<class ResultsT>
struct VariableAccessor<ResultsT, Category::noColumn>
```
Most functions have 4 implementations (1 in the generic template, 3 in specializations), most are `static` functions.

These functions attempt to use cluster names in the "multiple columns" specialization. 
```cpp
    174:                results.variableCaption = VCardT::Multiple::Caption(i);
    194:                results.variableCaption = VCardType::Multiple::Caption(i);
    214:                results.variableCaption = VCardType::Multiple::Caption(i);
    380:                results.variableCaption = thermal.clusters[i]->name();
    399:            results.variableCaption = thermal.clusters[idx]->name();
    405:            results.variableCaption = renewable.clusters[idx]->name();
    573:            results.variableCaption = VCardT::Caption();
    585:            return results.variableCaption;
    600:            results.variableCaption = getCaption<VCardType>(results);
    614:            results.variableCaption = VCardType::Caption();
```

I think we should also call `getCaption` in `BuildAnnualSurveyReport` (l. 614).

Reply:
Actually, **BuildAnnualSurveyReport** (in **info.h**) is called only when printing the year by year district report for an **area** output variable. As you know, the new output is not related to an area, this function is not called from the new output variable class.